File: pset2/Vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
 
int shift (char c);
int main(int argc, string argv[])
{
  if(argc != 2)
  {
      printf ("Usage: ./caesar keyword\n");
      return 1;
  }
 
  int n = strlen (argv [1]);
  for (int i = 0; i < n; i ++)
  {
      if (!isalpha (argv[1][i]))
      {
          printf ("Usage: ./caesar keyword\n");
          return 1;
      }
  }
 string s = get_string ("plaintext: ");
 printf ("ciphertext: ");
 int key = 0;
 int pointer = 0;
 for (int k = 0; k < strlen (s); k++)
 {
     if ((s[k] >= 'A' && s[k] <= 'Z') || (s[k] >= 'a' && s[k] <= 'z'))
     {
         key = shift(argv[1][pointer]);
         if (pointer == strlen (argv[1])-1)
         {
             pointer = 0;
         }
         else
         {
             pointer++;
         }
        
         if (s[k]%32 + key%26 <= 26)
         {
             printf ("%c", s[k] + key%26);
         }
         else
         {
             printf ("%c", s[k] - 26 + key%26);
         }
     }
     else
     {
         printf ("%c", s[k]);
     }
 }
 printf ("\n");
}
   
int shift (char c)
{
    int key = 0;
    c = tolower (c);
    char alphabet[26] = {'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'};
    for (int i = 0; i < 26; i++)
    {
        if (c == alphabet [i])
        {
            key = i;
           
        }
    }
    return key;
}

File: pset6/vigenere/vigenere.py
from cs50 import get_string
import sys


def main():

    if len(sys.argv) != 2:
        print("Usage: python vigenere.py k")
        exit(1)

    k = sys.argv[1]

    if(not k.isalpha()):
        print("Usage: python vigenere.py k")
        exit(1)

    p = get_string("plaintext: ")
    print("ciphertext: ", end="")

    pointer = 0

    for i in p:
        if(i >= 'A' and i <= 'Z') or (i >= 'a' and i <= 'z'):
            key = shift(k[pointer])
            if pointer == len(k)-1:
                pointer = 0
            else:
                pointer += 1

            if(ord(i) % 32 + key <= 26):
                print(chr(ord(i) + key), end="")
            else:
                print(chr(ord(i)-26 + key), end="")
        else:
            print(i, end="")

    print()


def shift(c):
    key = 0
    c = c.lower()
    alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
    key = alphabet.index(c)
    return key


if __name__ == "__main__":
    main()